config/runtime.cfg

```diff
@@ -217,6 +217,43 @@
 #          Example:  COB_CURRENT_DATE "2026/03/16 16:40:52"
 #                    current_date YYYYMMDDHHMMSS+01:00
 
+# Environment name:  COB_PROF_FILE
+#   Parameter name:  prof_file
+#          Purpose:  to define where COBOL profiling output should go
+#             Type:  string where $$ is replaced by process id, $b executable
+#                    basename, $f executable filename, $d date yyyymmdd,
+#                    $t time hhmmss
```

Comment:
I see that the type referred to as `string` in `runtime.cfg` is actually three types internally : ENV_STR, ENV_FILE, ENV_PATH. Maybe we should expose those explicitly in `runtime.cfg` ?

I notice that all those types of strings allow for `$$` and such replacements, yet they make sense only for some strings ; should we have another type of string, for instance "basic-string" and "extended-string", the first one not allowing expansion of `$$` and such ? Or do we just rely on the documentation to tell the user what he's allowed to do with those variables ?

Reply:
If every "string" is of this type (which makes the code easier) I'd leave it as "string" in runtime.cfg and in the introduction parts explain all the types, and for "string" add the possible replacements there once; afterwards we can then just reference them in the default/example values and expect users to know what this means.

Reply:
Right.

Interrestingly, `COB_DISPLAY_PRINT_FILE` and `COB_DISPLAY_PUNCH_FILE` are "declared" of type `ENV_STR` in `common.c`, though they are obviously paths (`ENV_PATH`). Should I change that (although unrelated to this PR) ?

Also, just noticing, but `COB_TRACE_FILE`, `COB_DUMP_FILE` and `COB_PROF_FILE` are defined as `ENV_FILE`, which should prevent from using path separators, yet the examples given do use path separators. Should those be `ENV_PATH` instead ?

Reply:
`ENV_PATH` allows path separators (= multiple paths, like in `COB_LIBRARY_PATH`), so `ENV_FILE` is corect for files. And yes the two filenames that are paths but are only string should be changed to `ENV_FILE`

Reply:
Hmm, one last : `COB_FILE_PATH` is defined as `ENV_PATH` ; shouldn't it be `ENV_FILE` ?

Reply:
It was defined as file before. In general we do want `COB_FILE_PATH` to be a path, but I don't know if someone did the merge of this from trunk (it shouldn't be that complicated). If that's in then runtime.cfg should also have a sample using multiple paths.

If it isn't: please set to FILE for now as it drops strange errors, if it is supported as PATH then we should have  matching documentation and test case.

Reply:
Maybe I'm mistaken, but as I understand that's how it is used (fileio.c) :
```
		if (cobsetptr->cob_file_path) {
			snprintf (file_open_buff, (size_t)COB_FILE_MAX, "%s%c%s",
				  cobsetptr->cob_file_path, SLASH_CHAR, src);
			file_open_buff[COB_FILE_MAX] = 0;
			strncpy (file_open_name, file_open_buff,
				 (size_t)COB_FILE_MAX);
		}
```
Used this way, I don't see how this could be a list of paths ?

Reply:
Ah, I see ; in trunk there is code to actually split COB_FILE_PATH...

Reply:
> That's the old code, so it cannot be a path list currently. Please adjust the common.c definition to match the current code.
> 
> BTW: You can do a lot of people a favor with a follow-up PR that does the backport of this feature. ;-)

I'll check that if that's not too much work ;)